File: Bit Manipulation/1780.Check if Number is a Sum of Powers of Three/Solution.java
class Solution {
    public boolean checkPowersOfThree(int n) {
        for (int i = 0; i < 1 << 15; i++) {
            int v = 0;
            for (int j = 0; j < 15; j++) {
                if ((i >> j & 1) == 1) {
                    v += Math.pow(3, j);
                    if (v > n) {
                        break;
                    }
                }
            }
            if (v == n) {
                return true;
            }
        }
        return false;
    }
}

File: Bit Manipulation/1780.Check if Number is a Sum of Powers of Three/Solution.py
class Solution:
    def checkPowersOfThree(self, n: int) -> bool:
        for i in range(1 << 15):
            v = 0
            for j in range(15):
                if i >> j & 1:
                    v += 3 ** j
                if v > n:
                    break
            if v == n:
                return True
        return False
